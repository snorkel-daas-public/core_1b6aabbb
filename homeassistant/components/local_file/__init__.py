"""The local_file component."""

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_FILE_PATH, Platform
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ConfigEntryError

from .const import DOMAIN
from .util import check_file_path_access

PLATFORMS = [Platform.CAMERA]


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up Local file from a config entry."""
    file_path: str = entry.options[CONF_FILE_PATH]
    if not await hass.async_add_executor_job(check_file_path_access, file_path):
        raise ConfigEntryError(
            translation_domain=DOMAIN,
            translation_key="not_readable_path",
            translation_placeholders={"file_path": file_path},
        )

    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload Local file config entry."""
    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
