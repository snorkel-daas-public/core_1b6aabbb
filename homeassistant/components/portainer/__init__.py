"""The Portainer integration."""

from __future__ import annotations

from pyportainer import Portainer

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import (
    CONF_API_KEY,
    CONF_API_TOKEN,
    CONF_HOST,
    CONF_URL,
    CONF_VERIFY_SSL,
    Platform,
)
from homeassistant.core import HomeAssistant
from homeassistant.helpers.aiohttp_client import async_create_clientsession

from .coordinator import PortainerCoordinator

_PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR]

type PortainerConfigEntry = ConfigEntry[PortainerCoordinator]


async def async_setup_entry(hass: HomeAssistant, entry: PortainerConfigEntry) -> bool:
    """Set up Portainer from a config entry."""

    client = Portainer(
        api_url=entry.data[CONF_URL],
        api_key=entry.data[CONF_API_TOKEN],
        session=async_create_clientsession(
            hass=hass, verify_ssl=entry.data[CONF_VERIFY_SSL]
        ),
    )

    coordinator = PortainerCoordinator(hass, entry, client)
    await coordinator.async_config_entry_first_refresh()

    entry.runtime_data = coordinator
    await hass.config_entries.async_forward_entry_setups(entry, _PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: PortainerConfigEntry) -> bool:
    """Unload a config entry."""
    return await hass.config_entries.async_unload_platforms(entry, _PLATFORMS)


async def async_migrate_entry(hass: HomeAssistant, entry: PortainerConfigEntry) -> bool:
    """Migrate old entry."""

    if entry.version < 2:
        data = dict(entry.data)
        data[CONF_URL] = data.pop(CONF_HOST)
        data[CONF_API_TOKEN] = data.pop(CONF_API_KEY)
        hass.config_entries.async_update_entry(entry=entry, data=data, version=2)

    return True
