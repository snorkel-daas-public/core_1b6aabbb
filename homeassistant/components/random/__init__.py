"""The random component."""

from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up a config entry."""
    await hass.config_entries.async_forward_entry_setups(
        entry, (entry.options["entity_type"],)
    )
    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload a config entry."""
    return await hass.config_entries.async_unload_platforms(
        entry, (entry.options["entity_type"],)
    )
