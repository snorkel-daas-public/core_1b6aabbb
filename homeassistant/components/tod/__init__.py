"""The Times of the Day integration."""

from __future__ import annotations

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up Times of the Day from a config entry."""
    await hass.config_entries.async_forward_entry_setups(
        entry, (Platform.BINARY_SENSOR,)
    )

    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload a config entry."""
    return await hass.config_entries.async_unload_platforms(
        entry, (Platform.BINARY_SENSOR,)
    )
