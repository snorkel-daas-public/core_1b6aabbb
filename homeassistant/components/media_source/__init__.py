"""The media_source integration."""

from __future__ import annotations

from typing import Protocol

from homeassistant.components import websocket_api
from homeassistant.core import HomeAssistant
from homeassistant.helpers import config_validation as cv
from homeassistant.helpers.integration_platform import (
    async_process_integration_platforms,
)
from homeassistant.helpers.typing import ConfigType

from . import http, local_source
from .const import (
    DOMAIN,
    MEDIA_CLASS_MAP,
    MEDIA_MIME_TYPES,
    MEDIA_SOURCE_DATA,
    URI_SCHEME,
    URI_SCHEME_REGEX,
)
from .error import MediaSourceError, Unresolvable
from .helper import async_browse_media, async_resolve_media
from .models import BrowseMediaSource, MediaSource, MediaSourceItem, PlayMedia

__all__ = [
    "DOMAIN",
    "MEDIA_CLASS_MAP",
    "MEDIA_MIME_TYPES",
    "BrowseMediaSource",
    "MediaSource",
    "MediaSourceError",
    "MediaSourceItem",
    "PlayMedia",
    "Unresolvable",
    "async_browse_media",
    "async_resolve_media",
    "generate_media_source_id",
    "is_media_source_id",
]


CONFIG_SCHEMA = cv.empty_config_schema(DOMAIN)


class MediaSourceProtocol(Protocol):
    """Define the format of media_source platforms."""

    async def async_get_media_source(self, hass: HomeAssistant) -> MediaSource:
        """Set up media source."""


def is_media_source_id(media_content_id: str) -> bool:
    """Test if identifier is a media source."""
    return URI_SCHEME_REGEX.match(media_content_id) is not None


def generate_media_source_id(domain: str, identifier: str) -> str:
    """Generate a media source ID."""
    uri = f"{URI_SCHEME}{domain or ''}"
    if identifier:
        uri += f"/{identifier}"
    return uri


async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up the media_source component."""
    hass.data[MEDIA_SOURCE_DATA] = {}
    http.async_setup(hass)

    # Local sources support
    await _process_media_source_platform(hass, DOMAIN, local_source)
    hass.http.register_view(local_source.UploadMediaView)
    websocket_api.async_register_command(hass, local_source.websocket_remove_media)

    await async_process_integration_platforms(
        hass, DOMAIN, _process_media_source_platform
    )
    return True


async def _process_media_source_platform(
    hass: HomeAssistant,
    domain: str,
    platform: MediaSourceProtocol,
) -> None:
    """Process a media source platform."""
    source = await platform.async_get_media_source(hass)
    hass.data[MEDIA_SOURCE_DATA][domain] = source
    if isinstance(source, local_source.LocalSource):
        hass.http.register_view(local_source.LocalMediaView(hass, source))
