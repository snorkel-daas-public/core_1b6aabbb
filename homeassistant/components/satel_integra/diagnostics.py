"""Diagnostics support for Satel Integra."""

from __future__ import annotations

from typing import Any

from homeassistant.components.diagnostics import async_redact_data
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_CODE
from homeassistant.core import HomeAssistant

TO_REDACT = {CONF_CODE}


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for the config entry."""
    diag: dict[str, Any] = {}

    diag["config_entry_data"] = dict(entry.data)
    diag["config_entry_options"] = async_redact_data(entry.options, TO_REDACT)

    diag["subentries"] = dict(entry.subentries)

    return diag
