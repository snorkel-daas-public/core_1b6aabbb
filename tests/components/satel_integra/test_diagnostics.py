"""Tests for satel integra diagnostics."""

from unittest.mock import AsyncMock

from syrupy.assertion import SnapshotAssertion
from syrupy.filters import props

from homeassistant.core import HomeAssistant

from tests.common import MockConfigEntry
from tests.components.diagnostics import get_diagnostics_for_config_entry
from tests.typing import ClientSessionGenerator


async def test_diagnostics(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    hass_client: ClientSessionGenerator,
    mock_config_entry_with_subentries: MockConfigEntry,
    mock_satel: AsyncMock,
) -> None:
    """Test diagnostics for config entry."""
    mock_config_entry_with_subentries.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_config_entry_with_subentries.entry_id)
    await hass.async_block_till_done()

    diagnostics = await get_diagnostics_for_config_entry(
        hass, hass_client, mock_config_entry_with_subentries
    )
    assert diagnostics == snapshot(exclude=props("created_at", "modified_at", "id"))
