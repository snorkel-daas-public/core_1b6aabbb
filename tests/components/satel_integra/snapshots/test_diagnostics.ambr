# serializer version: 1
# name: test_diagnostics
  dict({
    'config_entry_data': dict({
      'host': '192.168.0.2',
      'port': 7094,
    }),
    'config_entry_options': dict({
      'code': '**REDACTED**',
    }),
    'subentries': dict({
      'ID_OUTPUT': dict({
        'data': dict({
          'name': 'Output 1',
          'output_number': 1,
          'type': 'safety',
        }),
        'subentry_id': 'ID_OUTPUT',
        'subentry_type': 'output',
        'title': 'Output 1',
        'unique_id': 'output_1',
      }),
      'ID_PARTITION': dict({
        'data': dict({
          'arm_home_mode': 1,
          'name': 'Home',
          'partition_number': 1,
        }),
        'subentry_id': 'ID_PARTITION',
        'subentry_type': 'partition',
        'title': 'Home',
        'unique_id': 'partition_1',
      }),
      'ID_SWITCHABLE_OUTPUT': dict({
        'data': dict({
          'name': 'Switchable Output 1',
          'switchable_output_number': 1,
        }),
        'subentry_id': 'ID_SWITCHABLE_OUTPUT',
        'subentry_type': 'switchable_output',
        'title': 'Switchable Output 1',
        'unique_id': 'switchable_output_1',
      }),
      'ID_ZONE': dict({
        'data': dict({
          'name': 'Zone 1',
          'type': 'motion',
          'zone_number': 1,
        }),
        'subentry_id': 'ID_ZONE',
        'subentry_type': 'zone',
        'title': 'Zone 1',
        'unique_id': 'zone_1',
      }),
    }),
  })
# ---
